src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -45,6 +47,12 @@ public Envelope processEnvelope(byte[] metadataStream, String containerName) thr
         Envelope envelope = EntityParser.parseEnvelopeMetadata(inputStream);
         envelope.setContainer(containerName);
 
+        envelopeRepository
+            .findByContainerAndZipFileName(containerName, envelope.getZipFileName())
+            .ifPresent(dbEnvelope -> {
```

Comment:
I think you will also need to check the envelope status. If envelope status is DOC_UPLOAD_FAILURE then you should skip this as blob is not deleted when it fails to upload to DM store.